src/Traits/DetectsChanges.php

```diff
@@ -50,6 +58,20 @@ public function attributeValuesToBeLogged(string $processingEvent): array
 
     public static function logChanges(Model $model): array
     {
-        return collect($model)->only($model->attributesToBeLogged())->toArray();
+        return collect($model->attributesToBeLogged())->mapWithKeys(
+            function ($value) use ($model) {
+                if (is_iterable($value)) {
+                    foreach ($value as $methodCall) {
+                        $model = $model->$methodCall;
+                    }
+
+                    return [implode('.', $value) => $model];
+                } else {
```

Comment:
Generally I try to avoid `else` statements. Since you're using an early return in the `if` branch, you can just not use and `else`statement and put the second `return` after closing the `if` branch.